fuzz/fuzz_regexp.c

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2020 Google Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#include "libregexp.h"
+#include "quickjs.h"
+#include "quickjs-libc.h"
+
+#include <stdint.h>
+#include <stdio.h>
+
+#define CAPTURE_COUNT_MAX 255
+
+FILE *outfile=NULL;
+JSRuntime *rt;
+JSContext *ctx;
+
+int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+    if (outfile == NULL) {
```

Comment:
There is no way for outfile to not be null, what's the deal here?

Reply:
Honestly, I only used and improved the eval and compile targets, while the regexp target was simply copied from [oss-fuzz](https://github.com/google/oss-fuzz/blob/master/projects/quickjs/fuzz_regexp.c). But, I think the purpose of outfile is similar as of the `initialized` variable in the other two targets. If I simplify libFuzzer, then it's nothing more than a fuzzing harness with an infinite loop which calls the `LLVMFuzzerTestOneInput` function linked to it until it crashes. Since in a fuzzing session everything is about performance, the one-time initializations are preferred to be executed only once. So, I think the NULL-check of `outfile` also serves this purpose: after the first test run, `outfile` won't be NULL (most probably), hence the initialization won't run again either.

Reply:
On the other hand, I don't see the purpose of writing the result of `lre_exec` into this `outfile`, since it's never read back and file IO (especially unnecessary IO) and fuzzing are not the best friends (performance reasons). Could it be reasoned with some compiler hack to avoid optimizing out `lre_*` calls if their result is not used?

Reply:
No idea TBH.